### PR TITLE
feature (ref DPLAN163):

### PR DESCRIPTION
**Pull request by jalalifardemos**
### Ticket: 
DPLAN-163

Description: 
- create methods for statistics in StatementService.php
- use these methods in ProcedureStatisticsRpcController.php and StatementResourceType.php

### Linked PRs (optional)
List other PRs that are somehow connected to this and explain the connection. Don't
link private repositories in public ones, but the other way around is fine.

- https://github.com/demos-europe/demosplan-core/pull/3402

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

**Review by sierrazamudiodemos — CHANGES_REQUESTED** [review]

**Comment by sierrazamudiodemos** [comment]
I think the method where the stn status is calculated looks good! I added some comments

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php [hunk 1]
```diff
@@ -4571,6 +4574,51 @@ public function addMissingSortKeys($sort, string $defaultPropertyName, string $d
...
+
+        return 'processing';
+    }
+
+    public function getStatisticsOfProcedure(ProcedureInterface $procedure)
+    {
+        /** @var StatementInterface $statementsOfProcedure */
+        $statementsOfProcedure = $procedure->getStatements();
+        $statistics = [
+            'new'        => 0,
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php]: done in [1d00b0d](https://github.com/demos-europe/demosplan-core/pull/3417/commits/1d00b0dad8fba8200a1bd539c85574fc8d3dec49)

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php [hunk 2]
```diff
@@ -278,19 +278,8 @@ protected function getProperties(): array|ResourceConfigBuilderInterface
...
-                            return 'processing';
-                        } else {
-                            return 'completed';
-                        }
-                    }
-                }
-
-                return 'new';
-            });
+                $this->statementService->computetStatementStatus($statement);
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php]: done in [1d00b0d](https://github.com/demos-europe/demosplan-core/pull/3417/commits/1d00b0dad8fba8200a1bd539c85574fc8d3dec49)

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php [hunk 3]
```diff
@@ -278,19 +278,8 @@ protected function getProperties(): array|ResourceConfigBuilderInterface
...
-                            return 'processing';
-                        } else {
-                            return 'completed';
-                        }
-                    }
-                }
-
-                return 'new';
-            });
+                $this->statementService->getProcessintStatus($statement);
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/StatementResourceType.php]: done in [0f0ae42](https://github.com/demos-europe/demosplan-core/pull/3417/commits/0f0ae4240bb9817e433bd32b5256855d06d0bc14)

**Review by salisdemos — APPROVED** [review]

**Review by sierrazamudiodemos — APPROVED** [review]
thanks!

**Pull request merged**